File: backend/src/schemas/alert.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class AlertSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    fileId: str
    level: str
    message: str
    createdAt: datetime


class AlertListCriteriaSchema(BaseModel):
    limit: int = Field(default=20, ge=1, le=100)
    offset: int = Field(default=0, ge=0)

File: frontend/src/entities/alert.ts
import { EColorVariants } from '@/app/enums/ui'
import { ELevelStatus } from '@/app/enums/alert'

export interface IAlert {
  id: number
  fileId: string
  level: ELevelStatus
  message: string
  createdAt: string
}

export interface IAlertCriteria {
  offset: number
  limit: number
}

export interface IAlertMap {
  id: number
  fileId: string
  level: {
    variant: EColorVariants
    label: ELevelStatus
  }
  message: string
  createdAt: string
}

export interface IAlertState {
  criteria: IAlertCriteria
}

// CONSTANTS

export const ALERT_STATE: IAlertState = {
  criteria: { offset: 0, limit: 2 },
}

export const ALERT_HEADERS: IHeader[] = [
  { key: 'id', title: 'ID', classes: '' },
  { key: 'fileId', title: 'File ID', classes: '' },
  { key: 'level', title: 'Уровень', classes: '' },
  { key: 'message', title: 'Сообщение', classes: '' },
  { key: 'createdAt', title: 'Создан', classes: '' },
]
